tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs

```diff
@@ -172,25 +252,24 @@ public void FinishedSessionsAreSent()
             // then
             httpClient.Received(3).SendBeaconRequest(clientIp, Arg.Any<byte[]>());
             context.Received(1).HandleStatusResponse(statusResponse);
-            Assert.That(finishedSessions, Is.Empty);
         }
 
         [Test]
-        public void UnsuccessfulFinishedSessionsAreMovedBackToCache()
+        public void UnsuccessfulFinishedSessionsAreNotRemovedFromCache()
         {
             //given
             var target = new BeaconSendingCaptureOnState();
 
-            var finishedSession = CreateValidSession("127.0.0.1");
-            context.GetNextFinishedSession().Returns(finishedSession);
+            var finishedSession = new SessionWrapper(CreateValidSession("127.0.0.1"));
+            finishedSession.UpdateBeaconConfiguration(new BeaconConfiguration(1, DataCollectionLevel.OFF, CrashReportingLevel.OFF));
+            finishedSessions.Add(finishedSession);
             httpClient.SendBeaconRequest(Arg.Any<string>(), Arg.Any<byte[]>()).Returns((StatusResponse)null);
 
-
             //when calling execute
             target.Execute(context);
 
-            context.Received(1).GetNextFinishedSession();
-            context.Received(1).PushBackFinishedSession(finishedSession);
+            var tmp = context.Received(1).FinishedAndConfiguredSessions;

```

Comment:
`tmp` is never used

Reply:
Unfortunately an NSubstitute limitation http://nsubstitute.github.io/help/received-calls/